spectrum/checks.py

```diff
@@ -596,19 +601,6 @@ def profile(self, path):
         return annotation_links, pager_links
 
 
-    def _persistently_get(self, url):
```

Comment:
May not be applicable here. But just in case you're not aware, requests has some retry capabilities built in, I am configuring it for example here: https://github.com/elifesciences/peerscout/blob/develop/peerscout/utils/requests.py#L4

Reply:
We have some particular needs (like retrying 504 but not 50X in general, etc), but I'll keep it in mind, thanks.

Reply:
Just an additional note that you can (or indeed have to) configure the status codes to retry on. Probably not necessary here, but one of the nice features is that it supports an increasing backoff, apart from the maximum number of retries of course.
(It's very useful for particular external APIs frequently returning a 500)